src/plugins/saved_objects/public/save_modal/show_saved_object_save_modal.tsx

```diff
@@ -34,34 +33,42 @@ export function showSaveModal(
   saveModal: React.ReactElement<MinimalSaveModalProps>,
   Wrapper?: FC<PropsWithChildren<unknown>>
 ) {
-  const container = document.createElement('div');
-  const closeModal = () => {
-    ReactDOM.unmountComponentAtNode(container);
-    document.body.removeChild(container);
-    saveModal.props.onClose?.();
-  };
+  // initialize variable that will hold reference for unmount
+  // eslint-disable-next-line prefer-const
+  let unmount: ReturnType<ReturnType<typeof toMountPoint>>;
 
-  const onSave = saveModal.props.onSave;
+  const mount = toMountPoint(
+    React.createElement(function createSavedObjectModal() {
+      const closeModal = () => {
+        unmount();
+        // revert control back to caller after cleaning up modal
+        setTimeout(() => {
+          saveModal.props.onClose?.();
+        }, 0);
+      };
 
-  const onSaveConfirmed: MinimalSaveModalProps['onSave'] = async (...args) => {
-    const response = await onSave(...args);
-    // close modal if we either hit an error or the saved object got an id
-    if (Boolean(isSuccess(response) ? response.id : response.error)) {
-      closeModal();
-    }
-    return response;
-  };
-  document.body.appendChild(container);
-  const element = React.cloneElement(saveModal, {
-    onSave: onSaveConfirmed,
-    onClose: closeModal,
-  });
+      const onSave = saveModal.props.onSave;
 
-  const I18nContext = getI18n().Context;
-  ReactDOM.render(
-    <KibanaRenderContextProvider analytics={getAnalytics()} i18n={getI18n()} theme={getTheme()}>
-      <I18nContext>{Wrapper ? <Wrapper>{element}</Wrapper> : element}</I18nContext>
-    </KibanaRenderContextProvider>,
-    container
+      const onSaveConfirmed: MinimalSaveModalProps['onSave'] = async (...args) => {
+        const response = await onSave(...args);
+        // close modal if we either hit an error or the saved object got an id
+        if (Boolean(isSuccess(response) ? response.id : response.error)) {
+          closeModal();
+        }
+        return response;
+      };
+
+      const augmentedElement = React.cloneElement(saveModal, {
+        onSave: onSaveConfirmed,
+        onClose: closeModal,
+      });
+
+      return React.createElement(Wrapper ?? React.Fragment, {
+        children: augmentedElement,
+      });
+    }),
+    { analytics: getAnalytics(), theme: getTheme(), i18n: getI18n() }
   );
+
+  unmount = mount(document.createElement('div'));
```

Comment:
Would have opted to pass this mount definition to the `openModal` overlay provided by core, but it doesn't work well in scenarios where multiple modals are open, any previous open modal gets lost.

Reply:
Which scenarios cause multiple modals to open during save?

Reply:
The instance I encountered isn't related to dashboard itself, but rather in discover. This same method is used to trigger the save modal within discover, and in that scenario should a user need to create a new tag to be attached to the search that should be saved, ideally another modal is rendered that gets superimposed on the content this method would have rendered. However when the `openModal` overlay utils is used here because the caller for the create tag modal also uses `openModal` from core, when the create tag modal is rendered it discards the save modal with no way to recall it after the tag modal gets closed.